src/main/java/algorithms/curated170/hard/smallestrectangleenclosingblackpixels/SmallestRectangleEnclosingBlackPixelsDFS.java

```diff
@@ -0,0 +1,46 @@
+package algorithms.curated170.hard.smallestrectangleenclosingblackpixels;
+
+public class SmallestRectangleEnclosingBlackPixelsDFS {
+    
+    int left, right, bottom, top;
+    int[][] directions = new int[][] { { 1, 0 }, { -1, 0 }, { 0, 1 }, { 0, -1 } };
+
+    public int minArea(char[][] image, int x, int y) {
+        left = 1000000;
+        right = 0;
+        bottom = 1000000;
+        top = 0;
+        findRectangleBoundaries(image, x, y);
+        return (right - left + 1) * (top - bottom + 1);
+    }
+
+    private void findRectangleBoundaries(char[][] image, int x, int y) {
+        if (isOutOfBoundariesOrBlack(image, x, y)) {
+            return;
+        }
+
+        image[x][y] = '0';
+
+        for (int[] dir : directions) {
+            int nx = x + dir[0];
+            int ny = y + dir[1];
+            findRectangleBoundaries(image, nx, ny);
+        }
+        if (x < left) {
+            left = x;
+        }
+        if (y < bottom) {
+            bottom = y;
+        }
+        if (y > top) {
+            top = y;
+        }
+        if (x > right) {
+            right = x;
+        }
```

Comment:
Although programmatically it is not a problem, it would be better to keep the coordinate notation right:
  ```suggestion
   if (y < left) {
            left = y;
        }
        if (x < bottom) {
            bottom = x;
        }
        if (x > top) {
            top = x;
        }
        if (y > right) {
            right = y;
        }

```

Reply:
I think considering x to be vertical and y to be horizontal is less suitable.